### PR TITLE
update mouse position when receiving mouse wheel events

### LINKED ISSUE #14290: When Servo isn't the window directly in front, scroll is stuck in previously scrolled layer
On Mac, I've noticed that, when Servo isn't the window 'in front', then even though you can still scroll  Servo because it's still viewable, it is always the same 'layer' (in the sense of the Webrender layer) that scrolls, even if you move the cursor onto a different part of the page. 

So for example taking https://output.jsbin.com/lukiye, if you start by scrolling the main page, then move Servo to the back (for example take the terminal in front of it), while still showing Servo. You can still scroll the  main page, however when you move your cursor onto the div in the middle of it,  trying to scroll it will just scroll the page, not the child div. Same thing if you start by first scrolling the child div. 

In other words, the scrolling is 'stuck' in the layer that you happened to be scrolling the last time that Servo was still 'in front'. 

I've checked, and when Servo is no longer 'in front', the [scroll in webrender](https://github.com/servo/webrender/blob/master/webrender/src/frame.rs#L297) always receives the same 'cursor', even when you actually move your cursor around...

I've also noticed that when Servo isn't in front on start-up, you cannot scroll at all.

### PR DESCRIPTION
**Pull request by gterzian**
Update mouse position when receiving mouse wheel events.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #14290 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14808)

### CODE REVIEW

**Comment by gterzian** [comment]
Ok so tests currently fails, because the glutin side, https://github.com/servo/glutin/pull/109, needs to land first, I guess?

**Comment by gterzian** [comment]
@paulrouget thanks, yes makes sense since this is also done in `Event::MouseInput`

File: ports/glutin/window.rs [hunk 1]
```diff
@@ -421,11 +421,14 @@ impl Window {
...
                     WindowEvent::MouseWindowMoveEventClass(TypedPoint2D::new(x as f32, y as f32)));
             }
-            Event::MouseWheel(delta, phase) => {
+            Event::MouseWheel(delta, phase, pos) => {
                 let (dx, dy) = match delta {
                     MouseScrollDelta::LineDelta(dx, dy) => (dx, dy * LINE_HEIGHT),
                     MouseScrollDelta::PixelDelta(dx, dy) => (dx, dy),
                 };
+                if let Some((x, y)) = pos {
+                  self.mouse_pos.set(Point2D::new(x, y));
```
Comment (pcwalton) [inline ports/glutin/window.rs]: nit: use 4 space indentation.

**Comment by jdm** [comment]
Update glutin to use 0.7, squash all the commits together, and this can merge!

**Comment by gterzian** [comment]
When I try `./mach cargo-updat -p servo-glutin`, I get lots of duplicate dependencies error, in all the package that have been updated. How to resolve this?

**Comment by jdm** [comment]
Could you post the output you get? You'll need to commit the changes to the Cargo.lock file and add them to this PR, too.

**Comment by gterzian** [comment]
@jdm

```
./mach cargo-updat -p servo-glutin
We're assuming the 'cargo-updat' command is 'cargo-update' and we're executing it for you.

    Updating registry `https://github.com/rust-lang/crates.io-index`
      Adding adler32 v0.3.0
      Adding byteorder v1.0.0
      Adding cocoa v0.6.0
      Adding core-graphics v0.5.0
      Adding deflate v0.7.2
      Adding image v0.12.1
      Adding png v0.6.2
    Removing servo-glutin v0.6.3
      Adding servo-glutin v0.6.6
      Adding servo-glutin v0.7.0
```
then later, when running `./mach test-tidy`:
```
 ./Cargo.lock:1: duplicate versions for package `servo-glutin`
	The following packages depend on version 0.6.6 from 'crates.io':
		servo-skia
	The following packages depend on version 0.7.0 from 'crates.io':
		glutin_app
./Cargo.lock:1: duplicate versions for package `image`
	The following packages depend on version 0.10.4 from 'crates.io':
		webdriver_server
		compositing
		net_traits
		script
	The following packages depend on version 0.12.1 from 'crates.io':
		servo-glutin
		servo-glutin
./Cargo.lock:1: duplicate versions for package `cocoa`
	The following packages depend on version 0.5.2 from 'crates.io':
		embedding
	The following packages depend on version 0.6.0 from 'crates.io':
		servo-glutin
		servo-glutin
./Cargo.lock:1: duplicate versions for package `png`
	The following packages depend on version 0.5.2 from 'crates.io':
		image
	The following packages depend on version 0.6.2 from 'crates.io':
		image
./Cargo.lock:1: duplicate versions for package `core-graphics`
	The following packages depend on version 0.4.2 from 'crates.io':
		azure
		cocoa
		core-text
		gfx
		webrender
		webrender_traits
	The following packages depend on version 0.5.0 from 'crates.io':
		cocoa
		servo-glutin
		servo-glutin
./Cargo.lock:1: duplicate versions for package `byteorder`
	The following packages depend on version 0.5.3 from 'crates.io':
		bincode
		gfx
		image
		immeta
		jpeg-decoder
		mp4parse
		ogg
		ogg_metadata
		wayland-window
		webrender
		webrender_traits
		websocket
	The following packages depend on version 1.0.0 from 'crates.io':
		deflate
		image
  Progress: 100% (3/3)
`./mach test-tidy` failed: please fix the errors above
```

**Comment by Ms2ger** [comment]
That's done in #15111.

**Comment by gterzian** [comment]
@Ms2ger thanks.

I rebased off the latest, and this time also tried a `./mach cargo-update -a`, but I'm still getting some conflicts:

```
./Cargo.lock:1: duplicate versions for package `servo-glutin`
	The following packages depend on version 0.6.6 from 'crates.io':
		servo-skia
	The following packages depend on version 0.7.0 from 'crates.io':
		glutin_app
./Cargo.lock:1: duplicate versions for package `cocoa`
	The following packages depend on version 0.5.2 from 'crates.io':
		embedding
	The following packages depend on version 0.6.0 from 'crates.io':
		servo-glutin
		servo-glutin
./Cargo.lock:1: duplicate versions for package `core-graphics`
	The following packages depend on version 0.4.2 from 'crates.io':
		azure
		cocoa
		core-text
		gfx
		webrender
		webrender_traits
	The following packages depend on version 0.5.0 from 'crates.io':
		cocoa
		servo-glutin
		servo-glutin
```

**Comment by jdm** [comment]
I recommend not using `./mach cargo-update -a`. We will need https://github.com/servo/skia/pull/124 and a new version of skia published, and then update to use that. That should address the cases where we have multiple versions of glutin in use.

**Comment by gterzian** [comment]
ok thanks @jdm 

so as it currently stands, when only updating `servo-glutin`, these are the dups:

```
./Cargo.lock:1: duplicate versions for package `servo-glutin`
	The following packages depend on version 0.6.6 from 'crates.io':
		servo-skia
	The following packages depend on version 0.7.0 from 'crates.io':
		glutin_app
./Cargo.lock:1: duplicate versions for package `cocoa`
	The following packages depend on version 0.5.2 from 'crates.io':
		embedding
	The following packages depend on version 0.6.0 from 'crates.io':
		servo-glutin
		servo-glutin
./Cargo.lock:1: duplicate versions for package `core-graphics`
	The following packages depend on version 0.4.2 from 'crates.io':
		azure
		cocoa
		core-text
		gfx
		webrender
		webrender_traits
	The following packages depend on version 0.5.0 from 'crates.io':
		cocoa
		servo-glutin
		servo-glutin
```

**Comment by jdm** [comment]
Hmm, I thought https://github.com/servo/servo/pull/15111 had merged already. Looks like we need to wait for it, since it sorts out all of those probems.

**Comment by mbrubeck** [comment]
I think this is be un-blocked now.  `./mach cargo update -p skia` should be sufficient to update the Cargo.lock file correctly.

**Comment by nox** [comment]
This is not unblocked yet, various packages need to be bumped.

**Comment by mbrubeck** [comment]
I submitted a rebased version of this PR as #15333.  Thank you!

**Comment by gterzian** [comment]
thanks!

**Pull request closed without merging**